### ISSUE TITLE
Hang when the host partition is full.

### ISSUE THREAD

**Issue by jief666** [acidanthera/bugtracker#1198]

**Comment by vit9696** [comment]
Well, I assume that's related to OC. Please provide more details. Current description makes a bit little sense. Who hangs? At what circumstances? Any logs? What is the desired behaviour? Why? Etc.

**Comment by vit9696** [comment]
We performed tests on a tier one Z170 platform here. Starting with a full partition and...

- file logging enabled produces 0 KB sized log files or none.
- SysReport enabled produces 0 KB sized ACPI tables or no files.
- making screenshots produces 0 KB screenshots or no files.

Like this:
![1](https://user-images.githubusercontent.com/4348897/95183767-1a8cec80-07cf-11eb-8431-8777fa55e49c.jpeg)

This gives no evidence of any error in OpenCore code and aligns with our understanding of the codebase. Most likely your tests were done on a version with some dirty hacks and alike. Many thanks to @grassybor for holding the investigation.

**Comment by jief666** [comment]
On the 3 screenshots, there is space left on partition 235KB 140KB and 139KB.
I've written in title "full".

**Comment by jief666** [comment]
as in 0 bytes left.

**Comment by vit9696** [comment]
What you have written in the title is not really a proper bugreport, as we have to guess what actually happened. Please provide more details with a proper log (e.g. made with a serial). Currently there is no evidence the issue exists.

**Comment by jief666** [comment]
Is that : "Hang when the host partition on which OC is launched from has 0 bytes free." better ?
🥇 :-D:-D:-D There is no log, because the partition is full (euh sorry, has 0 bytes left free).

Try before saying I'm not reporting proper... or don't.

**Comment by vit9696** [comment]
Well, cannot you enable serial logging to test that? The issue is, for us it boots just fine with 0 bytes free as well, I simply forgot to attach the second screenshot.

![sys](https://user-images.githubusercontent.com/4348897/95186947-3b574100-07d3-11eb-85f5-69f050ee3f42.jpeg)

**Comment by jief666** [comment]
Here is the log :
`00:000 00:000 OCFS: WriteFileData file->Write returned Volume Full
00:081 00:081 OCFS: WriteFileData file->Write returned Volume Full
00:180 00:098 OCFS: WriteFileData file->Write returned Volume Full
00:241 00:061 OCFS: WriteFileData file->Write returned Volume Full
00:352 00:110 OCFS: WriteFileData file->Write returned Volume Full
00:402 00:050 OCFS: WriteFileData file->Write returned Volume Full
00:452 00:050 OCFS: WriteFileData file->Write returned Volume Full
00:542 00:090 OCFS: WriteFileData file->Write returned Volume Full
00:592 00:049 OCFS: WriteFileData file->Write returned Volume Full
00:632 00:039 OCFS: WriteFileData file->Write returned Volume Full
00:682 00:050 OCFS: WriteFileData file->Write returned Volume Full
00:810 00:127 OCFS: WriteFileData file->Write returned Volume Full
00:882 00:072 OCFS: WriteFileData file->Write returned Volume Full
00:961 00:079 OCFS: WriteFileData file->Write returned Volume Full
01:111 00:150 OCFS: WriteFileData file->Write returned Volume Full
01:162 00:050 OCFS: WriteFileData file->Write returned Volume Full
01:221 00:059 OCFS: WriteFileData file->Write returned Volume Full
01:281 00:060 OCFS: WriteFileData file->Write returned Volume Full
01:320 00:039 OCFS: WriteFileData file->Write returned Volume Full
01:380 00:060 OCFS: WriteFileData file->Write returned Volume Full
01:450 00:069 OCFS: WriteFileData file->Write returned Volume Full
01:500 00:050 OCFS: WriteFileData file->Write returned Volume Full
01:541 00:040 OCFS: WriteFileData file->Write returned Volume Full
01:650 00:109 OCFS: WriteFileData file->Write returned Volume Full
01:700 00:049 OCFS: WriteFileData file->Write returned Volume Full
01:741 00:040 OCFS: WriteFileData file->Write returned Volume Full
01:771 00:029 OCFS: WriteFileData file->Write returned Volume Full
01:801 00:030 OCFS: WriteFileData file->Write returned Volume Full
01:841 00:039 OCFS: WriteFileData file->Write returned Volume Full
01:871 00:030 OCFS: WriteFileData file->Write returned Volume Full
01:880 00:008 OCFS: WriteFileData file->Write returned Volume Full
01:882 00:002 OCFS: WriteFileData file->Write returned Volume Full
01:891 00:008 OCFS: WriteFileData file->Write returned Volume Full
01:900 00:008 OCFS: WriteFileData file->Write returned Volume Full
01:903 00:002 OCFS: WriteFileData file->Write returned Volume Full
01:912 00:008 OCFS: WriteFileData file->Write returned Volume Full
01:920 00:008 OCFS: WriteFileData file->Write returned Volume Full
01:923 00:002 OCFS: WriteFileData file->Write returned Volume Full
01:931 00:008 OCFS: WriteFileData file->Write returned Volume Full
01:933 00:001 OCFS: WriteFileData file->Write returned Volume Full
01:941 00:008 OCFS: WriteFileData file->Write returned Volume Full
01:952 00:010 OCFS: WriteFileData file->Write returned Volume Full
01:961 00:009 OCFS: WriteFileData file->Write returned Volume Full
01:971 00:010 OCFS: WriteFileData file->Write returned Volume Full
01:981 00:009 OCFS: WriteFileData file->Write returned Volume Full
01:983 00:001 OCFS: WriteFileData file->Write returned Volume Full
01:991 00:008 OCFS: WriteFileData file->Write returned Volume Full
01:993 00:001 OCFS: WriteFileData file->Write returned Volume Full
02:001 00:008 OCFS: WriteFileData file->Write returned Volume Full
02:010 00:008 OCFS: WriteFileData file->Write returned Volume Full
02:012 00:001 OCFS: WriteFileData file->Write returned Volume Full
02:020 00:008 OCFS: WriteFileData file->Write returned Volume Full
02:022 00:001 OCFS: WriteFileData file->Write returned Volume Full
02:030 00:008 OCFS: WriteFileData file->Write returned Volume Full
02:032 00:001 OCFS: WriteFileData file->Write returned Volume Full
02:040 00:008 OCFS: WriteFileData file->Write returned Volume Full
02:042 00:001 OCFS: WriteFileData file->Write returned Volume Full
02:051 00:008 OCFS: WriteFileData file->Write returned Volume Full
02:060 00:008 OCFS: WriteFileData file->Write returned Volume Full
02:061 00:001 OCFS: WriteFileData file->Write returned Volume Full
02:070 00:008 OCFS: WriteFileData file->Write returned Volume Full
02:072 00:002 OCFS: WriteFileData file->Write returned Volume Full
02:080 00:008 OCFS: WriteFileData file->Write returned Volume Full
02:082 00:001 OCFS: WriteFileData file->Write returned Volume Full
02:091 00:008 OCFS: WriteFileData file->Write returned Volume Full
02:092 00:001 OCFS: WriteFileData file->Write returned Volume Full
02:101 00:008 OCFS: WriteFileData file->Write returned Volume Full
02:103 00:001 OCFS: WriteFileData file->Write returned Volume Full
02:111 00:008 OCFS: WriteFileData file->Write returned Volume Full
02:120 00:008 OCFS: WriteFileData file->Write returned Volume Full
02:122 00:001 OCFS: WriteFileData file->Write returned Volume Full
02:131 00:008 OCFS: WriteFileData file->Write returned Volume Full
02:132 00:001 OCFS: WriteFileData file->Write returned Volume Full
02:141 00:008 OCFS: WriteFileData file->Write returned Volume Full
02:143 00:001 OCFS: WriteFileData file->Write returned Volume Full
02:152 00:009 OCFS: WriteFileData file->Write returned Volume Full
02:161 00:009 OCFS: WriteFileData file->Write returned Volume Full
02:170 00:008 OCFS: WriteFileData file->Write returned Volume Full
02:171 00:001 OCFS: WriteFileData file->Write returned Volume Full
02:180 00:008 OCFS: WriteFileData file->Write returned Volume Full
02:182 00:001 OCFS: WriteFileData file->Write returned Volume Full
02:190 00:008 OCFS: WriteFileData file->Write returned Volume Full
02:192 00:001 OCFS: WriteFileData file->Write returned Volume Full
02:201 00:008 OCFS: WriteFileData file->Write returned Volume Full
02:202 00:001 OCFS: WriteFileData file->Write returned Volume Full
02:211 00:008 OCFS: WriteFileData file->Write returned Volume Full
02:213 00:001 OCFS: WriteFileData file->Write returned Volume Full
02:221 00:008 OCFS: WriteFileData file->Write returned Volume Full
02:230 00:008 OCFS: WriteFileData file->Write returned Volume Full
02:231 00:001 OCFS: WriteFileData file->Write returned Volume Full
02:240 00:008 OCFS: WriteFileData file->Write returned Volume Full
02:242 00:001 OCFS: WriteFileData file->Write returned Volume Full
02:250 00:008 OCFS: WriteFileData file->Write returned Volume Full
02:252 00:001 OCFS: WriteFileData file->Write returned Volume Full
02:260 00:008 OCFS: WriteFileData file->Write returned Volume Full
02:262 00:001 OCFS: WriteFileData file->Write returned Volume Full
02:271 00:008 OCFS: WriteFileData file->Write returned Volume Full
02:272 00:001 OCFS: WriteFileData file->Write returned Volume Full
02:281 00:008 OCFS: WriteFileData file->Write returned Volume Full
02:283 00:001 OCFS: WriteFileData file->Write returned Volume Full
02:291 00:008 OCFS: WriteFileData file->Write returned Volume Full
02:301 00:009 OCFS: WriteFileData file->Write returned Volume Full
02:302 00:001 OCFS: WriteFileData file->Write returned Volume Full
02:311 00:008 OCFS: WriteFileData file->Write returned Volume Full
02:313 00:001 OCFS: WriteFileData file->Write returned Volume Full
02:321 00:008 OCFS: WriteFileData file->Write returned Volume Full
02:323 00:001 OCFS: WriteFileData file->Write returned Volume Full
02:331 00:008 OCFS: WriteFileData file->Write returned Volume Full
02:340 00:008 OCFS: WriteFileData file->Write returned Volume Full
02:342 00:001 OCFS: WriteFileData file->Write returned Volume Full
02:350 00:008 OCFS: WriteFileData file->Write returned Volume Full
02:352 00:001 OCFS: WriteFileData file->Write returned Volume Full
02:361 00:008 OCFS: WriteFileData file->Write returned Volume Full
02:363 00:001 OCFS: WriteFileData file->Write returned Volume Full
02:371 00:008 OCFS: WriteFileData file->Write returned Volume Full
02:373 00:001 OCFS: WriteFileData file->Write returned Volume Full
02:381 00:008 OCFS: WriteFileData file->Write returned Volume Full
02:390 00:008 OCFS: WriteFileData file->Write returned Volume Full
02:392 00:001 OCFS: WriteFileData file->Write returned Volume Full
02:400 00:008 OCFS: WriteFileData file->Write returned Volume Full
02:402 00:001 OCFS: WriteFileData file->Write returned Volume Full
02:410 00:008 OCFS: WriteFileData file->Write returned Volume Full
02:412 00:001 OCFS: WriteFileData file->Write returned Volume Full
02:421 00:008 OCFS: WriteFileData file->Write returned Volume Full
02:422 00:001 OCFS: WriteFileData file->Write returned Volume Full
02:431 00:008 OCFS: WriteFileData file->Write returned Volume Full
02:440 00:008 OCFS: WriteFileData file->Write returned Volume Full
02:442 00:001 OCFS: WriteFileData file->Write returned Volume Full
02:450 00:008 OCFS: WriteFileData file->Write returned Volume Full
02:452 00:001 OCFS: WriteFileData file->Write returned Volume Full
02:461 00:008 OCFS: WriteFileData file->Write returned Volume Full
02:462 00:001 OCFS: WriteFileData file->Write returned Volume Full
02:471 00:008 OCFS: WriteFileData file->Write returned Volume Full
02:480 00:008 OCFS: WriteFileData file->Write returned Volume Full
02:482 00:001 OCFS: WriteFileData file->Write returned Volume Full
02:490 00:008 OCFS: WriteFileData file->Write returned Volume Full
02:492 00:002 OCFS: WriteFileData file->Write returned Volume Full
02:500 00:008 OCFS: WriteFileData file->Write returned Volume Full
02:502 00:001 OCFS: WriteFileData file->Write returned Volume Full
02:511 00:008 OCFS: WriteFileData file->Write returned Volume Full
02:512 00:001 OCFS: WriteFileData file->Write returned Volume Full
02:521 00:008 OCFS: WriteFileData file->Write returned Volume Full
02:523 00:001 OCFS: WriteFileData file->Write returned Volume Full
02:531 00:008 OCFS: WriteFileData file->Write returned Volume Full
02:540 00:008 OCFS: WriteFileData file->Write returned Volume Full
02:541 00:001 OCFS: WriteFileData file->Write returned Volume Full
02:550 00:008 OCFS: WriteFileData file->Write returned Volume Full
02:552 00:001 OCFS: WriteFileData file->Write returned Volume Full
02:560 00:008 OCFS: WriteFileData file->Write returned Volume Full
02:562 00:001 OCFS: WriteFileData file->Write returned Volume Full
02:571 00:008 OCFS: WriteFileData file->Write returned Volume Full
02:572 00:001 OCFS: WriteFileData file->Write returned Volume Full
02:581 00:008 OCFS: WriteFileData file->Write returned Volume Full
02:590 00:008 OCFS: WriteFileData file->Write returned Volume Full
02:592 00:002 OCFS: WriteFileData file->Write returned Volume Full
02:600 00:008 OCFS: WriteFileData file->Write returned Volume Full
02:602 00:001 OCFS: WriteFileData file->Write returned Volume Full
02:611 00:008 OCFS: WriteFileData file->Write returned Volume Full
02:612 00:001 OCFS: WriteFileData file->Write returned Volume Full
02:621 00:008 OCFS: WriteFileData file->Write returned Volume Full
02:623 00:001 OCFS: WriteFileData file->Write returned Volume Full
02:631 00:008 OCFS: WriteFileData file->Write returned Volume Full
02:640 00:008 OCFS: WriteFileData file->Write returned Volume Full
02:642 00:001 OCFS: WriteFileData file->Write returned Volume Full
02:650 00:008 OCFS: WriteFileData file->Write returned Volume Full
02:652 00:001 OCFS: WriteFileData file->Write returned Volume Full
02:660 00:008 OCFS: WriteFileData file->Write returned Volume Full
02:662 00:001 OCFS: WriteFileData file->Write returned Volume Full
02:671 00:008 OCFS: WriteFileData file->Write returned Volume Full
02:672 00:001 OCFS: WriteFileData file->Write returned Volume Full
02:681 00:008 OCFS: WriteFileData file->Write returned Volume Full
02:690 00:008 OCFS: WriteFileData file->Write returned Volume Full
02:692 00:001 OCFS: WriteFileData file->Write returned Volume Full
02:700 00:008 OCFS: WriteFileData file->Write returned Volume Full
02:702 00:001 OCFS: WriteFileData file->Write returned Volume Full
02:710 00:008 OCFS: WriteFileData file->Write returned Volume Full
02:712 00:001 OCFS: WriteFileData file->Write returned Volume Full
02:721 00:008 OCFS: WriteFileData file->Write returned Volume Full
02:722 00:001 OCFS: WriteFileData file->Write returned Volume Full
02:732 00:009 OCFS: WriteFileData file->Write returned Volume Full
02:740 00:008 OCFS: WriteFileData file->Write returned Volume Full
02:742 00:001 OCFS: WriteFileData file->Write returned Volume Full
02:751 00:008 OCFS: WriteFileData file->Write returned Volume Full
02:760 00:009 OCFS: WriteFileData file->Write returned Volume Full
02:762 00:001 OCFS: WriteFileData file->Write returned Volume Full
02:770 00:008 OCFS: WriteFileData file->Write returned Volume Full
02:780 00:009 OCFS: WriteFileData file->Write returned Volume Full
02:782 00:001 OCFS: WriteFileData file->Write returned Volume Full
02:791 00:009 OCFS: WriteFileData file->Write returned Volume Full
02:800 00:008 OCFS: WriteFileData file->Write returned Volume Full
02:802 00:002 OCFS: WriteFileData file->Write returned Volume Full
02:811 00:008 OCFS: WriteFileData file->Write returned Volume Full
02:821 00:010 OCFS: WriteFileData file->Write returned Volume Full
02:830 00:008 OCFS: WriteFileData file->Write returned Volume Full
02:832 00:002 OCFS: WriteFileData file->Write returned Volume Full
02:840 00:008 OCFS: WriteFileData file->Write returned Volume Full
02:842 00:001 OCFS: WriteFileData file->Write returned Volume Full
02:852 00:009 OCFS: WriteFileData file->Write returned Volume Full
02:860 00:008 OCFS: WriteFileData file->Write returned Volume Full
02:863 00:002 OCFS: WriteFileData file->Write returned Volume Full
02:872 00:009 OCFS: WriteFileData file->Write returned Volume Full
02:880 00:008 OCFS: WriteFileData file->Write returned Volume Full
02:883 00:002 OCFS: WriteFileData file->Write returned Volume Full
02:891 00:008 OCFS: WriteFileData file->`

**Comment by jief666** [comment]
Sorry for line ending, github seems to remove them...

**Comment by vit9696** [comment]
Hm, next time please use three backticks to past the log.

````
```
line1
line 2
```
````

How did you get this line? If you look closer this line is marked as `DEBUG_VERBOSE`:

```
    if (EFI_ERROR (Status)) {
      DEBUG ((DEBUG_VERBOSE, "OCFS: WriteFileData file->Write returned %r\n", Status));
    } else if (WrittenSize != Size) {
```

`DEBUG_VERBOSE` logging in OpenCore is not supposed to be enabled even in DEBUG builds, see [these lines](https://github.com/acidanthera/OpenCorePkg/blob/0.6.2/OpenCorePkg.dsc#L275-L287). In fact, verbose logging is left for specific kinds of debugging particular libraries with specific debug libraries. E.g. you can find verbose debugging even in runtime builds, where the only way to print it would be to NVRAM or serial.

So, to summarise this a bug in whatever local changes you made to OpenCore codebase. Next time please be more clear about the environment you have and make sure that the issues are reproducible on the original OpenCore version (tagged or master). Let me know if you have more questions.

**Comment by jief666** [comment]
I changed this :
`
gEfiMdePkgTokenSpaceGuid.PcdDebugPrintErrorLevel|0x80400042
gEfiMdePkgTokenSpaceGuid.PcdFixedDebugPrintErrorLevel|0x80400042
`
in dsc file. I'm so use to it that I forgot to mention it.
Good news is it won't happen to average users.
Ok.

**Comment by vit9696** [comment]
Well, like I said, it is not supported and you will get all kinds of issues with it.

**Comment by vit9696** [comment]
In fact, if you check the documentation, you will discover that `DEBUG_VERBOSE` is made exclusively for _future debugging_ and is not intended to be enabled for the end-users. Next time please read the documentation carefully as it will help you to avoid obvious mistakes:

<img width="596" alt="Снимок экрана 2020-10-06 в 13 59 31" src="https://user-images.githubusercontent.com/4348897/95193236-29c66700-07dc-11eb-9051-e4a23ebfff72.png">

**Comment by jief666** [comment]
Don't worry, I won't make the obvious mistakes I already made twice.
